### PR TITLE
react_autocomplete

### PR DESCRIPTION
**Pull request by TetianaMD**
[DEMO LINK](https://TetianaMD.github.io/react_autocomplete/)

### CODE REVIEW

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]
image
search doesn't work

File: src/components/Autocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+interface AutoCompleteProps {
+  onSelected: (person: Person | null) => void;
+}
+
+export const AutoComplete: React.FC<AutoCompleteProps> = ({ onSelected }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [showAllPeople, setShowAllPeople] = useState(false);
+
+  const applyQuery = useCallback(
+    debounce(() => setShowAllPeople(true), 300),
```
Comment (etojeDenys) [inline src/components/Autocomplete.tsx:15]: image

File: src/components/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+export const AutoComplete: React.FC<AutoCompleteProps> = ({ onSelected }) => {
+  const [inputValue, setInputValue] = useState('');
+  const [showAllPeople, setShowAllPeople] = useState(false);
+  const dropdownRef = useRef(null);
+
+  const applyQuery = useCallback(() => {
+    setShowAllPeople(true);
+  }, []);
+
+  const debouncedApplyQuery = debounce(applyQuery, 1000);
```
Comment (SanyaBratashchuk) [inline src/components/Autocomplete.tsx:21]: image

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,98 +1,29 @@
...
+  const handleSelectedPerson = (person: Person | null) => {
+    setSelectedPerson(person);
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {selectedPerson ? `${name} (${born} - ${died})` : 'No selected person'}
```
Comment (volodymyr-soltys97) [inline src/App.tsx:21]: Consider will be better if you move this logic to the variable and use it here

File: src/components/Autocomplete.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import React, {
+  useCallback, useMemo, useRef, useState,
+} from 'react';
+import debounce from 'lodash.debounce';
```
Comment (volodymyr-soltys97) [inline src/components/Autocomplete.tsx:4]: Fix it
image

File: src/components/Autocomplete.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      </div>
+
+      <div
+        className="dropdown-menu"
+        role="menu"
+        data-cy="suggestions-list"
+      >
+        {showAllPeople && (
+          <div className="dropdown-content">
+            {filteredPersons.length > 0
```
Comment (volodymyr-soltys97) [inline src/components/Autocomplete.tsx:76]:
```suggestion
            {!!filteredPersons.length
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
To improve:
When you click on a person from the list, nothing happens, you need to show the person in the input and change the text above
image

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Consider hiding the list of suggestions when the input is not focused

**Pull request closed without merging**